src/server.js

```diff
@@ -179,11 +179,11 @@ class Server extends EventEmitter {
 
                 if (handlerOpts.upstreamProxyUrlParsed) {
                     this.log(handlerOpts.id, 'Using HandlerTunnelChain');
-                    return this.handlerRun(new HandlerTunnelChain(handlerOpts));
+                    return chain(request, socket, head, handlerOpts, this);
                 }
 
                 this.log(handlerOpts.id, 'Using HandlerTunnelDirect');
-                return direct(request, socket, head);
+                return direct(request, socket, head, handlerOpts, this);
```

Comment:
looks like there are just the 3 parameters on `direct`?

Reply:
Fixed.

Reply:
now it has 4 params, and you are feeding 5 (and server is 5th, not 4th), right?

Reply:
Ah, this one :) Now it's 5, `handlerOpts` won't be used but I prefer to keep this one so it looks the same like `chain`.